main.go

```diff
@@ -1,42 +1,121 @@
 package main
 
 import (
+	"encoding/base64"
+	"encoding/json"
+	"flag"
 	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 
+	"github.com/Strum355/log"
 	"github.com/bwmarrin/discordgo"
+	"github.com/hashicorp/consul/api"
+	consulwatch "github.com/hashicorp/consul/api/watch"
 )
 
-func getEnv(key string) (string, error) {
-	val, ok := os.LookupEnv(key)
-	if !ok {
-		return "", fmt.Errorf("Error finding %s", key)
-	}
-	return val, nil
+// Config represents the config pulled from Consul
+type Config struct {
+	PublicServer    string `json:"public"`
+	CommitteeServer string `json:"committee"`
 }
 
+var (
+	config     *Config
+	production bool
+)
+
 func main() {
-	token, err := getEnv("DISCORD_TOKEN")
-	if err != nil {
-		panic(err)
+	config = &Config{}
+	// Check for flags
+	production = *flag.Bool("p", false, "enables production with json logging")
+	flag.Parse()
+	if production {
+		log.InitJSONLogger(&log.Config{Output: os.Stdout})
+	} else {
+		log.InitSimpleLogger(&log.Config{Output: os.Stdout})
 	}
+
+	// Consul config
+	consulConfig := api.DefaultConfig()
+	var err error
+	consulConfig.Address, err = getEnv("CONSUL_ADDRESS")
+	exitError(err)
+
+	readConfig(consulConfig)
+
+	// Discord connection
+	token, err := getEnv("DISCORD_TOKEN")
+	exitError(err)
 	session, err := discordgo.New("Bot " + token)
-	if err != nil {
-		panic("Couldn't init discord connection")
-	}
+	exitError(err)
 	// Open websocket
 	err = session.Open()
-	if err != nil {
-		panic(err)
-	}
+	exitError(err)
 
 	// Maintain connection until a SIGTERM, then cleanly exit
-	fmt.Println("Bot running")
+	log.Info("Bot is Running")
 	sc := make(chan os.Signal, 1)
 	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt, os.Kill)
 	<-sc
-	fmt.Println("Cleanly exiting")
+	log.Info("Cleanly exiting")
 	session.Close()
 }
+
+func readConfig(consulConfig *api.Config) {
+	// Connect to consul
+	// Listen for consul updates
+	params := map[string]interface{}{
+		"type": "key",
+		"key":  "discordbot/servers",
+	}
+	watch, err := consulwatch.Parse(params)
+	exitError(err)
+	watch.Handler = func(idx uint64, data interface{}) {
+		// Serialize data
+		buf, err := json.MarshalIndent(data, "", "    ")
```

Comment:
what does `data` look like here? can you do `fmt.Printf("%#v %T", data, data)`

Reply:
`&api.KVPair{Key:"discordbot/servers", CreateIndex:0xb, ModifyIndex:0xb, LockIndex:0x0, Flags:0x0, Value:[]uint8{0x7b, 0x22, 0x63, 0x6f, 0x6d, 0x6d, 0x69, 0x74, 0x74, 0x65, 0x65, 0x22, 0x3a, 0x20, 0x22, 0x73, 0x73, 0x22, 0x2c, 0x20, 0x22, 0x70, 0x75, 0x62, 0x6c, 0x69, 0x63, 0x22, 0x3a, 0x20, 0x22, 0x73, 0x73, 0x73, 0x22, 0x7d}, Session:"", Namespace:""} *api.KVPair`

Reply:
Ill change the parser

Reply:
you can do `kv := data.(*api.KVPair)` and then just do `kv.Value` and happy days